app.py

```diff
@@ -19,23 +20,58 @@ def exportBag():
     "sample.tsv" in its "data" sub-directory. It creates
     a new workspace in a FireCloud namespace, and uploads
     the data into it.
+    Aborts execution and returns HTTP body back to client if any 
+    request operation fails to succeed.
     """
     req_body = app.current_request.raw_body
     req_query_params = app.current_request.query_params
     req_headers = app.current_request.headers
     os.chdir('/tmp')
     with zipfile.ZipFile(io.BytesIO(req_body), 'r') as archive:
         archive.extractall()
-    payload = []
-    payload.append('/tmp/manifest/data/participant.tsv')
-    payload.append('/tmp/manifest/data/sample.tsv')
+    data = {
+        'participant': '/tmp/manifest/data/participant.tsv',
+        'sample': '/tmp/manifest/data/sample.tsv'
+    }
+    url = 'https://api.firecloud.org/api/workspaces'
     workspace = req_query_params['workspace']
     namespace = req_query_params['namespace']
-    url = 'https://api.firecloud.org/api/workspaces'
     auth = req_headers['Authorization']
-    manifest_io = ManifestIO(payload, url, workspace, namespace, auth)
-    statusCode = manifest_io.import_tsv_to_fc()
-    return Response(body=statusCode,
-                    status_code=200,
+    # Dict to hold status codes (or booleans) for successful operation.
+    statusCode = {
+        'status codes': {
+            'workspace exists': '',
+            'stood up workspace': '',
+            'participant upload': '',
+            'sample upload': ''
+        }
+    }
+    manifest_io = ManifestIO(data, url, workspace, namespace, auth, statusCode)
+    # Check whether the workspace exists in this namespace, otherwise create it.
+    if not manifest_io.workspace_exists():
+        response = manifest_io.standup_workspace()
+        if response.status_code == 201:
+            manifest_io.statusCode['status codes']['stood up workspace'] = \
+                response.status_code
+        else:
+            return Response(body=response.text,
+                            headers={'Content-Type': 'text/plain',
+                                     'Accept': 'application/json'})
+    else:
+        manifest_io.statusCode['status codes']['stood up workspace'] = False
+
+    # Order keys of dictionary data alphabetically to ensure that participant
+    # is uploaded before sample is uploaded.
+    data_ordered = OrderedDict(sorted(data.items(), key=lambda t: t[0]))
+    for i, (key, val) in enumerate(data_ordered.items()):
+        r = manifest_io.import_tsv_to_fc(key)
+        if r.status_code != 200:
+            return Response(body=r.text,
+                            headers={'Content-Type': 'text/plain',
+                                     'Accept': 'application/json'})
```

Comment:
Same as other comment -- don't set Accept header and get Content-Type from FC response.